### PR TITLE
accounts/abi: substitude arg%d to the range keyword

### LINKED ISSUE #25252: abigen error: Failed to generate ABI binding in Golang
#### System information

Geth version: v.1.10.20
OS & Version: ubuntu 20.04
Commit hash : 
Golang version: 1.17.11
solc version: 0.8.13
solidity version: 0.8.0

Attached zip is my contract source file written in solidiy, and its abi / bin file generated by solc,

then I run this command to generate ABI binding in Golang

`
./abigen --abi=Vote.abi --bin=Vote.bin --pkg=contracts --type=vote --out=vote.go
`

but the experted file vote.go has not been generated, and the console printed this message:

```
Fatal: Failed to generate ABI binding: 465:76: expected ), found 'range' (and 10 more errors)

// Code generated - DO NOT EDIT.
// This file is a generated binding and any manual changes will be lost.

package contracts

import (
	"math/big"
	"strings"
	"errors"

	ethereum "github.com/ethereum/go-ethereum"
	"github.com/ethereum/go-ethereum/accounts/abi"
	"github.com/ethereum/go-ethereum/accounts/abi/bind"
	"github.com/ethereum/go-ethereum/common"
	"github.com/ethereum/go-ethereum/core/types"
	"github.com/ethereum/go-ethereum/event"
)

// Reference imports to suppress errors if they are not otherwise used.
var (
	_ = errors.New
	_ = big.NewInt
	_ = strings.NewReader
	_ = ethereum.NotFound
	_ = bind.Bind
	_ = common.Big1
	_ = types.BloomLookup
	_ = event.NewSubscription
)
// and more output ......
```

but it is ok to generate ABI binding in Java with same abi/bin files.

#### Steps to reproduce the behaviour
extract the attached zip into github.com/go-ethereum/build/bin, and run:
```
./abigen --abi=Vote.abi --bin=Vote.bin --pkg=contracts --type=vote --out=vote.go
```

[Vote.zip](https://github.com/ethereum/go-ethereum/files/9053951/Vote.zip)

### PR DESCRIPTION
**Pull request by gballet**
Fixes #25252 by replacing everything that looks like a keyword that would pose problem as a parameter name with a generic `arg%d` name.

### CODE REVIEW

File: accounts/abi/bind/bind.go [hunk 1]
```diff
@@ -43,6 +43,28 @@ const (
...
+	case "var":
+	case "const":
+	case "type":
+	case "iota":
+	case "return":
+	case "struct":
+	default:
+		return false
+	}
+
```
Comment (rjl493456442) [inline accounts/abi/bind/bind.go:64]: Please remove this blank line

File: accounts/abi/bind/bind.go [hunk 2]
```diff
@@ -114,7 +136,7 @@ func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]
 			normalized.Inputs = make([]abi.Argument, len(original.Inputs))
 			copy(normalized.Inputs, original.Inputs)
 			for j, input := range normalized.Inputs {
-				if input.Name == "" {
+				if input.Name == "" || isKeyWord(input.Name) {
```
Comment (rjl493456442) [inline accounts/abi/bind/bind.go:139]: We should also apply this check in `Event`, Line 183.

File: accounts/abi/bind/bind.go (hunk 1 above)
Comment (rjl493456442) [reply accounts/abi/bind/bind.go:64]: Golang has 25 keywords in total, should we list all of them?

```
break     default      func    interface  select
case      defer        go      map        struct
chan      else         goto    package    switch
const     fallthrough  if      range      type
continue  for          import  return     var
```

Comment (gballet) [reply accounts/abi/bind/bind.go:64]: I was too lazy, but you are right I should insert them all

**Review by rjl493456442 — APPROVED** [review]
LGTM

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM

**Pull request merged**